### PR TITLE
feat(rule): :sparkles: update geolocation-precision rule to validate app-gps values

### PR DESCRIPTION
**Pull request by AMarcosCastelo**
### Summary

Update the geolocation-precision-processor to validate the app-gps values that are passed in metadata.

### Related links

https://app.clickup.com/t/3005225/CARROT-957

---

- [x] **I, the PR author, declare that this PR works as expected and does not break any service. I also declare that I gave my best to apply all of the best practices and that I'm leaving the code better than I found it.**

## Summary by CodeRabbit

- **New Features**
  - Introduced test scenarios for validating geolocation precision, affecting approval/rejection outcomes based on latitude and longitude attributes.
  
- **Bug Fixes**
  - Added latitude and longitude attributes to enhance the precision of geolocation metadata validation.

- **Documentation**
  - Updated test cases to include scenarios for geolocation precision processing.

- **Refactor**
  - Improved geolocation metadata checks and mapping functions for better accuracy and reliability.

### CODE REVIEW

File: libs/methodologies/bold/processors/src/geolocation-precision-processor.spec.ts [hunk 1]
```diff
@@ -169,6 +169,39 @@ describe('GeolocationPrecisionRuleProcessor', () => {
...
+            stubDocumentEvent({
+              name: DocumentEventName.OUTPUT,
+              relatedDocument: massValidationReference,
+            }),
+          ],
+        },
+      ],
+      scenario:
+        'should return APPROVED when the app-gps-latitude and app-gps-longitude are found and the geolocation precision is valid',
+    },
```
[the remark this answers is not in the document]
Comment (AMarcosCastelo) [reply libs/methodologies/bold/processors/src/geolocation-precision-processor.spec.ts:204]: Yes, I'd like some help with this.

File: libs/methodologies/bold/processors/src/geolocation-precision.helpers.ts [hunk 2]
```diff
@@ -107,3 +108,24 @@ export const compareAddresses = (
...
+  if (is<number>(gpsLatitude) && is<number>(gpsLongitude)) {
+    return {
+      ...event.address,
+      latitude: gpsLatitude,
+      longitude: gpsLongitude,
+    };
+  }
+
+  return event.address;
+};
```
[the remark this answers is not in the document]
Comment (AMarcosCastelo) [reply libs/methodologies/bold/processors/src/geolocation-precision.helpers.ts]: I made some changes in this commit: [feature/geolocation-precision-update 818c875] test(rule): :white_check_mark: update tests and implementation of geolocation-precision

File: libs/methodologies/bold/processors/src/geolocation-precision-processor.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (AMarcosCastelo) [reply libs/methodologies/bold/processors/src/geolocation-precision-processor.spec.ts:204]: I made some changes in this commit: [feature/geolocation-precision-update 818c875] test(rule): :white_check_mark: update tests and implementation of geolocation-precision

File: libs/methodologies/bold/processors/src/geolocation-precision-processor.spec.ts [hunk 3]
```diff
@@ -334,9 +373,109 @@ describe('GeolocationPrecisionRuleProcessor', () => {
...
+              relatedDocument: massValidationReference,
+            }),
+          ],
+        },
+        recyclerHomologationDocumentStub,
+      ],
+      resultComment:
+        ruleDataProcessor['ResultComment'].METADATA_GEOLOCATION_INVALID,
+      scenario:
+        'should return REKECTED when the app-gps-latitude and app-gps-longitude are invalid data types',
```
Comment (RafaPalau) [inline libs/methodologies/bold/processors/src/geolocation-precision-processor.spec.ts]:
```suggestion
        'should return REJECTED when the app-gps-latitude and app-gps-longitude are invalid data types',
```

**Review by RafaPalau — CHANGES_REQUESTED** [review]

**Review by cris-santos — DISMISSED** [review]

**Pull request merged**